functions/src/glean.ts

```diff
@@ -0,0 +1,228 @@
+import Glean from "@mozilla/glean/node";
+import PingEncryptionPlugin from "@mozilla/glean/plugins/encryption";
+import axios from "axios";
+import { Mutex, Semaphore, withTimeout } from "async-mutex";
+
+import * as rallyMetrics from "./generated/rally.js";
+import * as userMetrics from "./generated/user.js";
+import * as enrollmentMetrics from "./generated/enrollment.js";
+import * as unenrollmentMetrics from "./generated/unenrollment.js";
+import * as rallyPings from "./generated/pings.js";
+
+import {
+  Uploader,
+  UploadResult,
+  UploadResultStatus,
+} from "@mozilla/glean/uploader";
+
+const GLEAN_DEBUG_VIEW_TAG = "MozillaRally";
+const GLEAN_RALLY_APP_ID = process.env.NODE_ENV !== "test" ? "rally-core" : "test-app-id";
+const GLEAN_APP_DISPLAY_VERSION = "TODO_app_version";
+const GLEAN_ENCRYPTION_JWK = {
```

Comment:
I think this is fine for now (maybe permanently), if we end up moving study JWKs to Firestore it might make sense to move this too, but we can re-deploy this quickly so it's not the same risk as with extensions.

Reply:
good point; I've made a note of it in the Study Repo Options proposal doc